src/System.Management.Automation/security/SecuritySupport.cs

```diff
@@ -248,7 +248,7 @@ private static bool IsCurrentProcessLaunchedByGpScript()
                 while (currentProcess != null)
                 {
                     if (string.Equals(gpScriptPath,
-                            PsUtils.GetMainModule(currentProcess).FileName, StringComparison.OrdinalIgnoreCase))
+                            currentProcess.MainModule.FileName, StringComparison.OrdinalIgnoreCase))
```

Comment:
We need a conclusion from @PaulHigin and @TravisEz13 

Looking [the API implementation](https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L193-L203) I believe it is safe change - main module has already loaded.

Reply:
What do you mean by "main module has already loaded"?
We cannot use Environment.ProcessPath here because `currentProcess` refers to one of the parent processes of PowerShell (that variable is wrongly named).

Reply:
To get the main module the API enumerates all modules. Docs say the enumeration can be interrupted if an assembly is loading in the time or main module is still initialized. 
I guess it is safe to ignore this in the context. In any case, this is not a reliable and predictable solution since we cannot predict when an assembly will load.

Reply:
In any case, using Process.MainModule directly does not affect reliability since the .NET BCL has a [built-in retry logic for the edge case you mentioned](https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L204-L224).
It seems that `PsUtils.GetMainModule` was made before the retry logic was implemented in the .NET BCL.

Reply:
MainModule could be null if the process is not loaded:  
https://github.com/dotnet/runtime/blob/6d238511167bdbf8cd16103e84b6befb45c0e301/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L200
The question is if that is relavant.

Reply:
Reviewing the code, it seems to be handled, where we aren't using currentProcess.  I think we are okay.